File: src/test/java/me/rasztabiga/fridgy/products/service/dto/ProductUnitDTOTest.java
package me.rasztabiga.fridgy.products.service.dto;

import static org.assertj.core.api.Assertions.assertThat;

import me.rasztabiga.fridgy.products.web.rest.TestUtil;
import org.junit.jupiter.api.Test;

class ProductUnitDTOTest {

    @Test
    void dtoEqualsVerifier() throws Exception {
        TestUtil.equalsVerifier(ProductUnitDTO.class);
        ProductUnitDTO productUnitDTO1 = new ProductUnitDTO();
        productUnitDTO1.setId(1L);
        ProductUnitDTO productUnitDTO2 = new ProductUnitDTO();
        assertThat(productUnitDTO1).isNotEqualTo(productUnitDTO2);
        productUnitDTO2.setId(productUnitDTO1.getId());
        assertThat(productUnitDTO1).isEqualTo(productUnitDTO2);
        productUnitDTO2.setId(2L);
        assertThat(productUnitDTO1).isNotEqualTo(productUnitDTO2);
        productUnitDTO1.setId(null);
        assertThat(productUnitDTO1).isNotEqualTo(productUnitDTO2);
    }
}

File: src/test/kotlin/me/rasztabiga/fridgy/products/service/dto/ProductUnitDTOTest.kt
package me.rasztabiga.fridgy.products.service.dto

import me.rasztabiga.fridgy.products.web.rest.equalsVerifier
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ProductUnitDTOTest {

    @Test
    fun dtoEqualsVerifier() {
        equalsVerifier(ProductUnitDTO::class)
        val productUnitDTO1 = ProductUnitDTO()
        productUnitDTO1.id = 1L
        val productUnitDTO2 = ProductUnitDTO()
        assertThat(productUnitDTO1).isNotEqualTo(productUnitDTO2)
        productUnitDTO2.id = productUnitDTO1.id
        assertThat(productUnitDTO1).isEqualTo(productUnitDTO2)
        productUnitDTO2.id = 2L
        assertThat(productUnitDTO1).isNotEqualTo(productUnitDTO2)
        productUnitDTO1.id = null
        assertThat(productUnitDTO1).isNotEqualTo(productUnitDTO2)
    }
}
